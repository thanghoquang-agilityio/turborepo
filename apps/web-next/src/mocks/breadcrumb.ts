export const MOCK_BREADCRUMB_ITEMS = [
  { name: 'Products', link: 'products' },
  { name: 'Detail Product', link: '7' },
]
