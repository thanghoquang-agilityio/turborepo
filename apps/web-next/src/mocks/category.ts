export const MOCK_CATEGORY_LIST = [
  {
    name: 'Kid',
    value: 'kid',
    total: 18,
  },
  {
    name: 'Man',
    value: 'man',
    total: 12,
  },
  {
    name: 'Woman',
    value: 'woman',
    total: 23,
  },
  {
    name: 'Sport',
    value: 'sport',
    total: 34,
  },
]

export const MOCK_CATEGORY_RESPONSE = [
  {
    name: 'Kid',
    value: 'kid',
    products: {
      data: [],
    },
  },
  {
    name: 'Man',
    value: 'man',
    products: {
      data: [],
    },
  },
  {
    name: 'Woman',
    value: 'woman',
    products: {
      data: [],
    },
  },
  {
    name: 'Sport',
    value: 'sport',
    products: {
      data: [],
    },
  },
]
