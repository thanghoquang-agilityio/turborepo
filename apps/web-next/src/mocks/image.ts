import { SRC_IMAGE_NOT_AVAILABLE } from '@/constants'

export const MOCK_IMAGE = {
  data: {
    id: '24',
    attributes: {
      name: 'default',
      url: SRC_IMAGE_NOT_AVAILABLE,
    },
  },
}

export const MOCK_LIST_IMAGE_RESPONSE = [
  {
    name: 'default1',
    url: SRC_IMAGE_NOT_AVAILABLE,
  },
  {
    name: 'default2',
    url: SRC_IMAGE_NOT_AVAILABLE,
  },
  {
    name: 'default3',
    url: SRC_IMAGE_NOT_AVAILABLE,
  },
  {
    name: 'default4',
    url: SRC_IMAGE_NOT_AVAILABLE,
  },
]
