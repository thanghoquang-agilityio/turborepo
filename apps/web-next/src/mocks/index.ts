export * from './breadcrumb'
export * from './cart'
export * from './category'
export * from './image'
export * from './product'
export * from './radio-group'
export * from './user'
