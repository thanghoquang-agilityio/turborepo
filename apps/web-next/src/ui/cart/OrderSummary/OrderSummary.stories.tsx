import type { Meta, StoryObj } from '@storybook/react'

import OrderSummary from '../OrderSummary'

const meta = {
  title: 'Components/OrderSummary',
  tags: ['autodocs'],
  component: OrderSummary,
  parameters: {
    layout: 'centered',
  },
} satisfies Meta<typeof OrderSummary>

export default meta
type Story = StoryObj<typeof OrderSummary>

export const Default: Story = {
  args: {
    subTotal: 369,
    itemTotal: 3,
  },
  argTypes: {
    onPress: { action: 'checkout' },
  },
}
