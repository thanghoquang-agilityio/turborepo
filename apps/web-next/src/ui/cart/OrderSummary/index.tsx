import { memo } from 'react'

import { Textarea } from '@nextui-org/react'

// Components
import { Button, Text } from '@/components'
// Constants
import { borderStyle } from '@/constants'
// Utils
import { formatAmountCurrency } from '@/utils'

interface OrderSummaryProps {
  subTotal: number
  itemTotal: number
  onPress?: () => void
}

const OrderSummary = memo(
  ({ subTotal, itemTotal, onPress }: OrderSummaryProps) => {
    const formatSubTotal = formatAmountCurrency(subTotal.toString())
    const classBorder = `${borderStyle} before:top-[0px] before:left-[0px]`

    return (
      <div className='flex max-w-[305px] flex-col gap-4 bg-background-100'>
        <div className='w-full rounded-medium border-large border-background-300 bg-background-100'>
          <Text variant='title' size='lg' className='p-4 text-center'>
            Order Summary
          </Text>
          <div className={`relative flex justify-between p-4 ${classBorder}`}>
            <Text variant='primary' size='md' className='max-w-[100px]'>
              Sub Total
            </Text>
            <Text variant='primary' size='md' className='max-w-[100px]'>
              {formatSubTotal}
            </Text>
          </div>
          <div className={`relative flex justify-between p-4 ${classBorder}`}>
            <Text variant='primary' size='md' className='max-w-[100px]'>
              Items Total
            </Text>
            <Text variant='primary' size='md' className='max-w-[100px]'>
              {itemTotal}
            </Text>
          </div>
        </div>
        <Textarea
          label='Add a gift message to your order'
          labelPlacement='outside'
          classNames={{
            base: 'w-full p-4 bg-background-100 border-large border-background-300 rounded-medium',
            label:
              'text-primary-300 text-base font-semibold group-data-[filled-within=true]:text-primary-300',
            inputWrapper:
              'border-medium border-background-300 rounded-none mt-1 min-h-[100px]',
            input: 'text-primary-300 text-sm font-medium',
          }}
        />
        <Text variant='subTitle' size='sm' className='text-secondary-500'>
          Shipping & taxes calculated at checkout
        </Text>
        <Button
          variant='solid'
          size='lg'
          additionalClass='h-[50px]'
          onClick={onPress}
          aria-label='Go To Checkout'
        >
          Go To Checkout
        </Button>
      </div>
    )
  }
)

OrderSummary.displayName = 'OrderSummary'
export default OrderSummary
