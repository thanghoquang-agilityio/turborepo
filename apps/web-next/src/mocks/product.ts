import { ProductVariantModel } from '@/types'

import { MOCK_IMAGE } from './image'

// New flattened product variants for testing
export const MOCK_PRODUCT_VARIANTS: ProductVariantModel[] = [
  {
    id: 1,
    color: 'slate blue',
    size: 'small,medium,large,extra large,XXL',
    stock: '20,20,20,20,20',
    tax: 5,
  },
  {
    id: 2,
    color: 'blue',
    size: 'small,medium,large,extra large,XXL',
    stock: '20,20,20,20,20',
    tax: 5,
  },
]

export const MOCK_PRODUCT = {
  id: 1,
  documentId: 'mock-product-1',
  image: MOCK_IMAGE,
  name: 'TDX Sinkers',
  variants: 5,
  price: 675.0,
  star: 5,
  comments: 121,
  description: '',
  discount: 0,
  likes: 11,
  brand: 'MOCK BRAND',
  productVariants: MOCK_PRODUCT_VARIANTS,
}

export const MOCK_PRODUCTS_RESPONSE = [
  {
    id: 1,
    documentId: 'mock-product-1',
    name: 'TDX Sinkers',
    variants: 5,
    price: 675.0,
    star: 5,
    comments: 121,
    brand: 'MOCK BRAND',
    description: '',
    discount: 0,
    likes: 11,
    productVariants: [],
  },
  {
    id: 2,
    documentId: 'mock-product-2',
    name: 'TDX Sinkers',
    variants: 5,
    price: 675.0,
    star: 5,
    comments: 121,
    brand: 'MOCK BRAND',
    description: '',
    discount: 0,
    likes: 11,
    productVariants: [],
  },
  {
    id: 3,
    documentId: 'mock-product-3',
    name: 'TDX Sinkers',
    variants: 5,
    price: 675.0,
    star: 5,
    comments: 121,
    brand: 'MOCK BRAND',
    description: '',
    discount: 0,
    likes: 11,
    productVariants: [],
  },
  {
    id: 4,
    documentId: 'mock-product-4',
    name: 'TDX Sinkers',
    variants: 5,
    price: 675.0,
    star: 5,
    comments: 121,
    brand: 'MOCK BRAND',
    description: '',
    discount: 0,
    likes: 11,
    productVariants: MOCK_PRODUCT_VARIANTS,
    image: MOCK_IMAGE,
  },
]
