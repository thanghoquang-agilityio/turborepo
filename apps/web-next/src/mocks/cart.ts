import { CartItemResponse } from '@/types'
import { MOCK_PRODUCT_VARIANTS } from './product'

export const MOCK_CART_ITEM = {
  id: '1',
  variantId: '1',
  documentId: '1',
  name: 'Embrace Sideboard',
  color: 'wheat',
  size: 'small',
  quantityDefault: 2,
  stock: 4,
  price: 53,
  star: 1,
  comments: 1,
  tax: 1,
  discount: 1,
}

// New flattened cart response structure
export const MOCK_CART_RESPONSE: CartItemResponse[] = [
  {
    id: 1,
    documentId: 'cart-item-1',
    size: 'small',
    quantity: 2,
    createdAt: '2024-01-01T00:00:00.000Z',
    updatedAt: '2024-01-01T00:00:00.000Z',
    publishedAt: '2024-01-01T00:00:00.000Z',
    productVariantId: MOCK_PRODUCT_VARIANTS[0],
    userId: 1,
  },
  {
    id: 2,
    documentId: 'cart-item-2',
    size: 'large',
    quantity: 2,
    createdAt: '2024-01-01T00:00:00.000Z',
    updatedAt: '2024-01-01T00:00:00.000Z',
    publishedAt: '2024-01-01T00:00:00.000Z',
    productVariantId: MOCK_PRODUCT_VARIANTS[1],
    userId: 1,
  },
]

export const MOCK_ORDER_SUMMARY = {
  subTotal: 396,
  itemTotal: 3,
}
