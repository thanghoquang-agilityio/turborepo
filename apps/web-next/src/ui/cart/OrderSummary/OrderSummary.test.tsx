import { render } from '@testing-library/react'

import { MOCK_ORDER_SUMMARY } from '@/mocks'

import OrderSummary from '.'

describe('OrderSummary', () => {
  const renderComponent = () => render(<OrderSummary {...MOCK_ORDER_SUMMARY} />)

  it('should match snapshot', () => {
    const { container } = renderComponent()
    expect(container).toMatchSnapshot()
  })
})
