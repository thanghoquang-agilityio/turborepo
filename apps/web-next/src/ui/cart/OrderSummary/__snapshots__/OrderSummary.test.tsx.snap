// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderSummary should match snapshot 1`] = `
<div>
  <div
    class="flex max-w-[305px] flex-col gap-4 bg-background-100"
  >
    <div
      class="w-full rounded-medium border-large border-background-300 bg-background-100"
    >
      <p
        class="text-primary-200 dark:text-primary-300 font-semibold text-lg no-underline text-ellipsis whitespace-nowrap overflow-hidden p-4 text-center"
      >
        Order Summary
      </p>
      <div
        class="relative flex justify-between p-4 before:content-[''] before:absolute before:w-full before:border-t-small before:border-background-300 before:top-[0px] before:left-[0px]"
      >
        <p
          class="text-primary-300 font-semibold text-base no-underline text-ellipsis whitespace-nowrap overflow-hidden max-w-[100px]"
        >
          Sub Total
        </p>
        <p
          class="text-primary-300 font-semibold text-base no-underline text-ellipsis whitespace-nowrap overflow-hidden max-w-[100px]"
        >
          $396.00
        </p>
      </div>
      <div
        class="relative flex justify-between p-4 before:content-[''] before:absolute before:w-full before:border-t-small before:border-background-300 before:top-[0px] before:left-[0px]"
      >
        <p
          class="text-primary-300 font-semibold text-base no-underline text-ellipsis whitespace-nowrap overflow-hidden max-w-[100px]"
        >
          Items Total
        </p>
        <p
          class="text-primary-300 font-semibold text-base no-underline text-ellipsis whitespace-nowrap overflow-hidden max-w-[100px]"
        >
          3
        </p>
      </div>
    </div>
    <div
      class="group flex flex-col data-[hidden=true]:hidden w-full p-4 bg-background-100 border-large border-background-300 rounded-medium"
      data-has-elements="true"
      data-has-label="true"
      data-has-value="true"
      data-slot="base"
    >
      <label
        class="z-10 pointer-events-none origin-top-left rtl:origin-top-right subpixel-antialiased block relative will-change-auto !duration-200 !ease-out motion-reduce:transition-none transition-[transform,color,left,opacity] group-data-[filled-within=true]:pointer-events-auto pb-1.5 pe-2 max-w-full text-ellipsis overflow-hidden text-primary-300 text-base font-semibold group-data-[filled-within=true]:text-primary-300"
        data-slot="label"
        for="react-aria-:r0:"
        id="react-aria-:r1:"
      >
        Add a gift message to your order
      </label>
      <div
        class="relative w-full inline-flex tap-highlight-transparent flex-row items-center shadow-sm px-3 gap-3 bg-default-100 data-[hover=true]:bg-default-200 group-data-[focus=true]:bg-default-100 h-10 !h-auto transition-background motion-reduce:transition-none !duration-150 outline-none group-data-[focus-visible=true]:z-10 group-data-[focus-visible=true]:ring-2 group-data-[focus-visible=true]:ring-focus group-data-[focus-visible=true]:ring-offset-2 group-data-[focus-visible=true]:ring-offset-background py-2 border-medium border-background-300 rounded-none mt-1 min-h-[100px]"
        data-has-multiple-rows="true"
        data-slot="input-wrapper"
        style="cursor: text;"
      >
        <div
          class="inline-flex w-full h-full box-border items-start group-data-[has-label=true]:items-start"
          data-slot="inner-wrapper"
        >
          <textarea
            aria-label="Add a gift message to your order"
            aria-labelledby="react-aria-:r0: react-aria-:r1:"
            class="w-full bg-transparent !outline-none placeholder:text-foreground-500 focus-visible:outline-none data-[has-start-content=true]:ps-1.5 data-[has-end-content=true]:pe-1.5 file:cursor-pointer file:bg-transparent file:border-0 autofill:bg-transparent bg-clip-text resize-none data-[hide-scroll=true]:scrollbar-hide group-data-[has-value=true]:text-default-foreground transition-height !duration-100 motion-reduce:transition-none text-primary-300 text-sm font-medium"
            data-hide-scroll="true"
            data-slot="input"
            id="react-aria-:r0:"
          />
        </div>
      </div>
    </div>
    <p
      class="text-foreground-300 font-normal text-sm no-underline text-ellipsis whitespace-nowrap overflow-hidden text-secondary-500"
    >
      Shipping & taxes calculated at checkout
    </p>
    <button
      aria-label="Go To Checkout"
      class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap subpixel-antialiased overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none data-[hover=true]:opacity-hover gap-2.5 font-semibold bg-primary-300 text-background-100 py-5 xl:min-w-[310px] lg:min-w-[260px] sm:min-w-[310px] min-w-[160px] text-base rounded-[30px] h-[50px]"
      type="button"
    >
      Go To Checkout
    </button>
  </div>
</div>
`;
